src/main/kotlin/org/rust/lang/refactoring/generateConstructor/ui.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.refactoring.generateConstructor
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.project.Project
+import com.intellij.ui.SimpleColoredComponent
+import org.jetbrains.annotations.TestOnly
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.openapiext.isUnitTestMode
+import javax.swing.JTree
+
+class RsStructMemberChooserMember(val base: MemberChooserObjectBase, val member: ConstructorArguments) : ClassMember {
+    private val text: String = "${member.argumentIdentifier}  :   ${member.typeReference}"
+    override fun renderTreeNode(component: SimpleColoredComponent?, tree: JTree?) {
+        component?.append(text)
+    }
+
+    override fun getParentNodeDelegate(): MemberChooserObject? = base
+
+    override fun getText() = "${member.argumentIdentifier} : ${member.typeReference}"
+
+    override fun equals(other: Any?): Boolean {
+        return text == (other as? RsStructMemberChooserMember)?.text
+    }
+
+    fun formattedText() = text
+    override fun hashCode() = text.hashCode()
+}
+
+fun showConstructorArgumentsChooser(
+    structItem: RsStructItem,
+    project: Project
+): List<ConstructorArguments>? {
+
+    val base = MemberChooserObjectBase(structItem.name, structItem.getIcon(0))
+
+    val chooser = if (isUnitTestMode) {
+        MOCK!!
+    } else memberChooserDialog
+    return chooser(project,
+        ConstructorArguments.fromStruct(structItem).map { RsStructMemberChooserMember(base, it) })?.map { it.member }
+}
+typealias StructMemberChooser = (
+    project: Project,
+    all: List<RsStructMemberChooserMember>
+) -> List<RsStructMemberChooserMember>?
```

Comment:
You can use `chooser.selectedElements.orEmpty()` inside `memberChooserDialog` instead of using nullable `List` (look at `org/rust/lang/refactoring/implementMembers/ui.kt`).

Reply:
Null value and empty list has a different meaning. Null - user close selector, and empty list - user chose "selected none"

Reply:
IMO it is not obvious and unclear. Also, [similar Kotlin's action](https://upsource.jetbrains.com/kotlin/file/55ff519aa80da45cc99100d6789e309e398bc1b8/idea/src/org/jetbrains/kotlin/idea/actions/generate/KotlinGenerateSecondaryConstructorAction.kt) doesn't use nullable `List` and creates an empty constructor if a user closes the selector.

@vlad20012 how should the action work in this case?

Reply:
IMO, invoke action, if user close editor is unexpected for him